### PR TITLE
Fix notary keys requests for all keys

### PR DESCRIPTION
**Pull request by S7evinK**
This should be more spec compliant:
> If no key IDs are given to be queried, the notary server should query for all keys.

### CODE REVIEW

File: federationapi/federationapi_test.go [hunk 1]
```diff
@@ -362,3 +368,118 @@ func TestRoomsV3URLEscapeDoNot404(t *testing.T) {
...
+		httpBody      string
+		pubKeyRequest *gomatrixserverlib.PublicKeyNotaryLookupRequest
+		validateFunc  func(t *testing.T, response util.JSONResponse)
+	}{
+		{
+			name: "empty httpBody",
+			validateFunc: func(t *testing.T, resp util.JSONResponse) {
+				want := util.JSONResponse{
+					Code: http.StatusBadRequest,
+					JSON: spec.BadJSON("The request body could not be decoded into valid JSON. unexpected end of JSON input"),
```
Comment (kegsay) [inline federationapi/federationapi_test.go:386]: It would be nicer to assert the error code and not the entire error message, as that is pretty brittle.

File: federationapi/internal/query.go [hunk 2]
```diff
@@ -43,6 +43,15 @@ func (a *FederationInternalAPI) fetchServerKeysFromCache(
 	ctx context.Context, req *api.QueryServerKeysRequest,
 ) ([]gomatrixserverlib.ServerKeys, error) {
 	var results []gomatrixserverlib.ServerKeys
+
+	// We got a request for _all_ server keys, return them.
+	if len(req.KeyIDToCriteria) == 0 {
```
Comment (kegsay) [inline federationapi/internal/query.go:48]: Technically `{"server_keys":{"servera":null}}` will then be treated as getting all keys.

File: federationapi/routing/keys.go [hunk 3]
```diff
@@ -217,10 +221,7 @@ func NotaryKeys(
 		}
 	}
 
-	var response struct {
-		ServerKeys []json.RawMessage `json:"server_keys"`
-	}
-	response.ServerKeys = []json.RawMessage{}
```
Comment (kegsay) [inline federationapi/routing/keys.go:223]: Bad refactor. You forgot to do this, so no keys will be `server_keys: null` not `server_keys: []`.

**Review by kegsay — CHANGES_REQUESTED** [review]
Please add additional testing for the `null` vs `[]` edge case.
